ibis/backends/dask/execution/generic.py

```diff
@@ -0,0 +1,869 @@
+        return true
+    else:
+        # TODO double check this is the right way to do this
+        out = dd.from_array(np.repeat(false, len(true)))
+        out.index = true.index
+        return out
+
+
+# scalar, scalar, Series
+@execute_node.register(ops.Where, boolean_types, scalar_types, dd.Series)
+def execute_node_where_scalar_scalar_series(op, cond, true, false, **kwargs):
+    return dd.from_array(np.repeat(true, len(false))) if cond else false
+
+
+@execute_node.register(DaskTable, DaskClient)
+def execute_database_table_client(
+    op, client, timecontext: Optional[TimeContext], **kwargs
+):
+    df = client.dictionary[op.name]
+    if timecontext:
+        begin, end = timecontext
+        if TIME_COL not in df:
+            raise com.IbisError(
+                f'Table {op.name} must have a time column named {TIME_COL}'
+                ' to execute with time context.'
+            )
+        # filter with time context
+        mask = df[TIME_COL].between(begin, end)
+        return df.loc[mask].reset_index(drop=True)
+    return df
+
+
+@execute_node.register(ops.IfNull, dd.Series, simple_types)
+@execute_node.register(ops.IfNull, dd.Series, dd.Series)
+def execute_node_ifnull_series(op, value, replacement, **kwargs):
+    return value.fillna(replacement)
+
+
+@execute_node.register(ops.IfNull, simple_types, dd.Series)
+def execute_node_ifnull_scalar_series(op, value, replacement, **kwargs):
+    return (
+        replacement
+        if isnull(value)
+        else dd.from_pandas(
+            pd.Series(value, index=replacement.index), npartitions=1
+        )
+    )
+
+
+@execute_node.register(ops.NullIf, dd.Series, dd.Series)
+def execute_node_nullif_series(op, series1, series2, **kwargs):
+    return series1.where(series1 != series2)
+
+
+@execute_node.register(ops.NullIf, dd.Series, simple_types)
+def execute_node_nullif_series_scalar(op, series, value, **kwargs):
+    return series.where(series != value)
+
+
+@execute_node.register(ops.NullIf, simple_types, dd.Series)
+def execute_node_nullif_scalar_series(op, value, series, **kwargs):
+    # TODO - not preserving the index
+    return dd.from_array(da.where(series.eq(value).values, np.nan, value))
+
+
+def wrap_case_result(raw: np.ndarray, expr: ir.ValueExpr):
+    """Wrap a CASE statement result in a Series and handle returning scalars.
+
+    Parameters
+    ----------
+    raw : ndarray[T]
+        The raw results of executing the ``CASE`` expression
+    expr : ValueExpr
+        The expression from the which `raw` was computed
+
+    Returns
+    -------
+    Union[scalar, Series]
+    """
+    # TODO - improve this
+    raw_1d = np.atleast_1d(raw)
+    if np.any(isnull(raw_1d)):
+        result = dd.from_array(raw_1d)
+    else:
+        result = dd.from_array(
+            raw_1d.astype(constants.IBIS_TYPE_TO_PANDAS_TYPE[expr.type()])
+        )
+    # TODO - we force computation here
+    if isinstance(expr, ir.ScalarExpr) and result.size.compute() == 1:
```

Comment:
Why are we forcing computation here?

Reply:
We need to actually know the `result.size` to do the boolean test.

Reply:
kk sounds fine. We can look later. Should try to get rid of these computation in the complication step as much as possible.

Reply:
Agreed on the general point of aiming to not materialize as much as possible.